src/drivers/gps/gps.cpp

```diff
@@ -381,7 +404,10 @@ GPS::task_main()
 					_report_gps_pos.epv = 10000.0f;
 					_report_gps_pos.fix_type = 0;
 
-					if (!(_pub_blocked)) {
+					/* update mavlink hil gps parameter */
+					update_param_hil_gps();
```

Comment:
Yes, I fixed like this 
https://github.com/stmoon/Firmware/commit/f1c33262c8a7b3b7b44c637345e07f3d5d6cda84

Just only call param_find()  during initialization time in update_param_hil_gps() function .

Do you want to call the update_param_hil_gps()  during initialization time?

I thought parameter should be updated every time because it can be changed anytime.
However it is ok to call during initialization time just only once. :-)